malduck/procmem/procmem.py

```diff
@@ -118,26 +118,35 @@ def m(self):
 
     def close(self, copy=False):
         """
-        Closes opened files referenced by ProcessMemory object
+        Closes opened files referenced by ProcessMemory object owned by this object.
 
         If copy is False (default): invalidates the object.
 
         :param copy: Copy data into string before closing the mmap object (default: False)
         :type copy: bool
         """
-        if self.f is not None:
-            if self.mapped_memory is not None:
-                if copy:
-                    self.mapped_memory.seek(0)
-                    contents = self.mapped_memory.read()
-                    buf: Optional[bytearray] = bytearray(contents)
-                else:
-                    buf = None
-                self.mapped_memory.close()
-                self.mapped_memory = None
-                self.memory = buf
-            self.f.close()
-            self.f = None
+        if self.mapped_memory is None:
+            # Nothing to close
+            return
```

Comment:
Is there any case to legitimately trigger this?
If no, shouldn't it rather raise an exception?

Reply:
I don't see any advantage of throwing such exception. Even closing file doesn't throw anything if you try to close a file that is already closed.
```
In [4]: f = open(".mwdb")                                                       

In [5]: f.close()                                                               

In [6]: f.close()                                                               

In [7]: f.close()   
```